libbeat/processors/condition.go

```diff
@@ -32,6 +32,9 @@ type Condition struct {
 		name    string
 		filters map[string]match.Matcher
 	}
+	hasfields struct {
```

Comment:
1. Maybe better just have hasfields []string?

2. And maybe rename it to requiredFields? In this case would be more natural "requiredFields []string"

Reply:
`requiredFields` is indeed an elegant name, but I feel it doesn't go that well with `when` and other conditions like `not`. I think we should keep this discussion open and try to find a better name for the condition.